src/sched.rs

```diff
@@ -59,3 +60,61 @@ pub fn get_policy(pid: i32) -> Result<Policy, ()> {
         policy @ _ => panic!("Policy {} does not exist", policy)
     }
 }
+
+/// Set the cpu affinity for the current thread See `set_affinity()`.
+pub fn set_self_affinity(cpuset: CpuSet) -> Result<(), ()> {
+    set_affinity(0, cpuset)
+}
+
+/// Set the cpu affinity for a thread.
+pub fn set_affinity(pid: i32, cpuset: CpuSet) -> Result<(), ()> {
+    match unsafe { sched_setaffinity(pid, cpuset.len(), cpuset.mask_ptr()) } {
```

Comment:
I think that the unsafety of calling `sched_setaffinity` could be encapsulated in safe method on `CpuSet` that is just called here and on L#89